src/waveform/waveformwidgetfactory.cpp

```diff
@@ -1127,98 +1069,77 @@ WaveformWidgetAbstract* WaveformWidgetFactory::createWaveformWidget(
-        case WaveformWidgetType::GLSLRGBWaveform:
-            widget = new GLSLRGBWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::GLSLRGBStackedWaveform:
-            widget = new GLSLRGBStackedWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::GLVSyncTest:
-            widget = new GLVSyncTestWidget(viewer->getGroup(), viewer);
-            break;
-#ifdef MIXXX_USE_QOPENGL
-        case WaveformWidgetType::AllShaderRGBWaveform:
-            widget = new allshader::RGBWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderLRRGBWaveform:
-            widget = new allshader::LRRGBWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderFilteredWaveform:
-            widget = new allshader::FilteredWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderRGBStackedWaveform:
-            widget = new allshader::RGBStackedWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderSimpleWaveform:
-            widget = new allshader::SimpleWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderHSVWaveform:
-            widget = new allshader::HSVWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderTexturedFiltered:
-            widget = new allshader::WaveformWidgetTexturedFiltered(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderTexturedRGB:
-            widget = new allshader::WaveformWidgetTexturedRGB(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::AllShaderTexturedStacked:
-            widget = new allshader::WaveformWidgetTexturedStacked(viewer->getGroup(), viewer);
-            break;
-#else
-        case WaveformWidgetType::QtSimpleWaveform:
-            widget = new QtSimpleWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::QtWaveform:
-            widget = new QtWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::QtVSyncTest:
-            widget = new QtVSyncTestWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::QtHSVWaveform:
-            widget = new QtHSVWaveformWidget(viewer->getGroup(), viewer);
-            break;
-        case WaveformWidgetType::QtRGBWaveform:
-            widget = new QtRGBWaveformWidget(viewer->getGroup(), viewer);
-            break;
+        auto info = WaveformWidgetInfoBase::find(type);
+
+        if (info) {
+            // waveform widget types registered with WaveformWidgetInfo's
+            widget = info->create(viewer->getGroup(), viewer).release();
+        } else {
+            switch (type) {
```

Comment:
likewise, once WaveformWidgetInfo's have been implemented for all WaveformWidgetTypes this switch can be removed, and the creation of the WaveformWidget's is done with 
```
widget = info->create(viewer->getGroup(), viewer).release();
```
above.